docs/changes/table/15735.api.rst

```diff
@@ -0,0 +1,28 @@
+The order of columns as they appear in a ``Table`` constructed as ``Table(rows=...)``
+may differ between ``astropy`` ``6.0`` and ``6.1``. Specifically, in ``6.0``,
+then column ordering was determined by the first :class:`dict` if it contained values
+for all the columns, or by sorting the column names alphabetically if it did not.
+In ``6.1``, columns are simply inserted as they appear when the :class:`list` of
+:class:`dict` is iterated over.
+
+As an example,
+
+  >>> data = [{'b': 10, 'c': 7, },
+  ...         {'a': 15, 'c': 35, 'b': 20}]
+  >>> t = Table(rows=data)
+  >>> print(t)
+
+
+would print, in ``6.0``
+   a   b   c
+  --- --- ---
+   --  10   7
+   15  20  35
+
+
+and in ``6.1``
+
+   b   c   a
+  --- --- ---
+   10   7  --
+   20  35  15
```

Comment:
Changelog is usually limited to one line or a few ones, no examples... maybe a good hint that this deserves a whatnew section in 7.0 ?

Reply:
This is an interesting question since we need to communicate this API change effectively. Would this go in the What's New for 6.1? Is that too high-profile or just right?

In general, is there any technical or aesthetic reason to not have longer changelog entries with examples? I'm just wondering about the driver for the precedent.

Reply:
Currently whatsnew is probably the best place. 
Changelog entries used to be restricted to one line per PR, then with towncrier we relaxed a bit this constraint (nobody verifies it), but the current format is not really adequate to have long explanations mixed with one-line entries.  Also because it all goes in one big file.
I suggested in the APE22 discussions that we could improve this, maybe with some inspiration from pandas (e.g. https://pandas.pydata.org/docs/whatsnew/v2.1.0.html), to have a better presentation of changes and more room to highlight important ones. Maybe something to discuss.

Reply:
I'm confused, my understanding was that "what's new" pages were constructed from changelog sample files. Where should I put this note then ?

Reply:
THe what's new is hand-written: https://github.com/astropy/astropy/tree/main/docs/whatsnew

Reply:
If this needs What's New again, please re-apply the What's New needed label. Thanks!